.github/workflows/build-with-flang.yml

```
@@ -30,3 +30,4 @@ jobs:
         export FPM_FC=$FC
         export FPM_FFLAGS="-mmlir -allow-assumed-rank -O3"
         fpm test
+        (set +e ; stdout_stderr=$(fpm test --target false-assertion --flag -DRUN_FALSE_ASSERTIONS) ; echo "$stdout_stderr" | grep "expected 2; actual value is 1" &> /dev/null ; test $? = 0) 
```

IIUC the error stop is actually being written to stderr, so you also need to explicitly capture that, which does not happen by default with `$()`. I suspect that's why the CI is currently failing for this PR.

I also recommend ensuring the full output from the `fpm` invcation reaches the action log, so that when it fails you can see why. Using a temp file is the easiest way to do that.

Cosmetic nit: The trailing commands can potentially be simplified using the `grep -q` option.

```suggestion
        (set +e ; fpm test --target false-assertion --flag -DRUN_FALSE_ASSERTIONS 2>&1 | tee output ; grep -q "expected 2; actual value is 1" output) 
```

And of course the corresponding changes to the gfortran version.